src/chainstate/stacks/db/transactions.rs

```diff
@@ -592,9 +590,9 @@ pub mod test {
         let addr = auth.origin().address_testnet();
         let recv_addr = StacksAddress { version: 1, bytes: Hash160([0xff; 20]) };
 
-        let mut tx_stx_transfer = StacksTransaction::new(TransactionVersion::Testnet,
-                                                         auth.clone(),
-                                                         TransactionPayload::TokenTransfer(recv_addr.clone(), 123, TokenTransferMemo([0u8; 34])));
+        let mut tx_stx_transfer = StacksTransaction::new(
```

Comment:
Can you add a test for doing a token transfer with a contract address?  i.e. extend `process_token_transfer_stx_transaction()` so it covers both address variants?